plugins/continuous-toolbox/src/ContinuousToolbox.js

```diff
@@ -58,13 +70,58 @@ export class ContinuousToolbox extends Blockly.Toolbox {
   /** @override */
   updateFlyout_(_oldItem, newItem) {
     if (newItem) {
-      const target = this.flyout_.getCategoryScrollPosition(newItem.name_).y;
-      this.flyout_.scrollTarget = target;
-      this.flyout_.stepScrollAnimation();
+      const target = this.getFlyout()
+          .getCategoryScrollPosition(newItem.name_).y;
+      this.getFlyout().scrollTo(target);
+    }
+  }
+
+  /** @override */
+  shouldDeselectItem_(oldItem, newItem) {
+    // Should not deselect if the same category is clicked again.
+    return (oldItem && oldItem !== newItem);
+  }
+
+  /**
+   * Gets a category by name.
+   * @param {string} name Name of category to get.
+   * @return {?Blockly.ToolboxCategory} Category, or null if not
+   *    found.
+   */
+  getCategoryByName(name) {
+    const category = this.contents_.find(
+        (item) => item instanceof Blockly.ToolboxCategory &&
```

Comment:
Not for this PR but I think this means we should have the kind stored on the toolbox item.

Reply:
even if it were provided as a method or something, the type would still have to be refined here because getName is only available on the type ToolboxCategory. And just calling a method like isCategory() (if i'm understanding your comment correctly) wouldn't be enough to tell the compiler that the type of this is indeed ToolboxCategory.

Reply:
Ah, I thought since we were already having to cast to `Blockly.ToolboxCategory` below then the check from the compiler didn't matter up here. But that makes sense.